zerver/tests/test_events.py

```diff
@@ -708,6 +709,60 @@ def test_typing_events(self) -> None:
         )
         check_typing_stop("events[0]", events[0])
 
+    def test_stream_typing_events(self) -> None:
+        stream_id = get_stream("Denmark", self.user_profile.realm).id
+        topic = "streams typing"
+
+        events = self.verify_action(
+            lambda: do_send_stream_typing_notification(
+                self.user_profile,
+                "start",
+                stream_id,
+                topic,
+            ),
+            state_change_expected=False,
+        )
+        check_typing_start("events[0]", events[0])
+
+        events = self.verify_action(
+            lambda: do_send_stream_typing_notification(
+                self.user_profile,
+                "stop",
+                stream_id,
+                topic,
+            ),
+            state_change_expected=False,
+        )
+        check_typing_stop("events[0]", events[0])
+
+        # Having client_capability `stream_typing_notification=False`
+        # shouldn't produce any events.
+        events = self.verify_action(
+            lambda: do_send_stream_typing_notification(
+                self.user_profile,
+                "start",
+                stream_id,
+                topic,
+            ),
+            state_change_expected=False,
+            stream_typing_notifications=False,
+            num_events=0,
+        )
+        self.assertEqual(events, [])
+
+        events = self.verify_action(
+            lambda: do_send_stream_typing_notification(
+                self.user_profile,
+                "stop",
+                stream_id,
+                topic,
+            ),
+            state_change_expected=False,
+            stream_typing_notifications=False,
+            num_events=0,
+        )
+        self.assertEqual(events, [])
```

Comment:
Added these to test our new `client_capability` functionality. Is it a correct way to test something like this?